src/components/CatastropheToggle.vue

```diff
@@ -42,30 +110,47 @@ export default defineComponent({
         };
     },
     computed: {
-        catastropheToggles() {
-            // TODO: this should be done based on a prop.
-            // TODO: put reminder to update this in model enum
-            return [
-                // TODO: keep track of checked state internally
-                // TODO: use localized names
-                {count: 213, iconPath: '/icons/attention.png',
-                 pillClass: 'catastrophe-all', name: 'Tous', checked: true},
-                {count: 37, iconPath: '/icons/flood_w.png',
-                 pillClass: 'catastrophe-flood', name: 'Innondations',
-                 checked: true},
-                {count: 24, iconPath: '/icons/forest_fire_w.png',
-                 pillClass: 'catastrophe-forest-fire', name: 'Feux de forêt',
-                 checked: false},
-                {count: 0, iconPath: '/icons/violent_storm_w.png',
-                 pillClass: 'catastrophe-violent-storm', name: 'Orages violents',
-                 checked: true},
-                {count: 8, iconPath: '/icons/tornado_w.png',
-                 pillClass: 'catastrophe-tornado', name: 'Tornades',
-                 checked: true},
-                {count: 144, iconPath: '/icons/heat_wave_w.png',
-                 pillClass: 'catastrophe-heat-wave', name: 'Vagues de chaleur',
-                 checked: true},
-            ]
+        catastropheToggles(): List<Toggle> {
+            var toggles: Toggle[] = [];
+            toggles.push({
+                count: this.allCatastrophes.size,
+                iconPath: '/icons/attention.png',
+                pillClass: 'catastrophe-all',
+                name: this.$t('all_catastrophes'),
+                checked: this.filter.equals(allCatastrophesFilter()),
+                onChange: this.onFilterAllChange,
+            });
+            for (const toggle of TOGGLES) {
+                const count = this.allCatastrophes.filter(
```

Comment:
Vu que tu utilises une immutable list au lieu d'un array, tu peux faire `count()` direct au lieu de `filter()`  + `size`.

Alternativement, obtiens les comptes de tous les types en une shot avec un seul call à `countBy()` en dehors du for. Ça va t'éviter de parcourir N fois la liste.

Reply:
TIL countBy, merci, j'connais pas l'API d'immutable List :smile: